rfc/rfc-79/rfc-79.md

```diff
@@ -0,0 +1,108 @@
+
+
+## Background
+The table service operations compact and cluster are by default "non-cancellable" plans, meaning that once a plan is scheduled it will stay as as a pending instant until a caller invokes the table service execute API on the table service instant and completes it. Specifically, if an inflight execution fails, a rollback plan is implictly created and executed for the instant which will delete all new instant/data files, but will keep the table service plan (repeating until the instant is completed). The below visualization captures these transitions at a high level 
+
+If a non-cancellable table service is configured for async execution then each of the aforementioned instant transitions for execution can potentially be preformed by seperate concurrent jobs. Typically this is expected only if an execution attempt fails after performing a step and the next execution attempt performs a rollback and re-attempts creating data files and comitting the instant. But  
+
+
+
+### Concurrent writers can execute table service plan 
+
+When a writer executes a compact, cluster, or logcompaction plan, it will first rollback any existing inflight attempt, and (depending on the type and configuration of the table service) optionally re-execute it. This can lead to dataset corruption if one writer is rolling back the instant while another is still executing it. See https://issues.apache.org/jira/browse/HUDI-7503 for an example.
+Independent/outside of HUDI, a user may have an orchestration setup of concurrent writers where sometimes multiple writers can execute the sample plan simultaneously, due to a transient failure (at the orchestration leve) or misconfiguration. While it can be argued that HUDI does not to provide any guarantees of correctness if concurrent writers execute the same plan, updating HUDI to guard against this scenario would reduce the operational overhead and complexity of deploying a large-scale HUDI writer orchestration.
```

Comment:
> While it can be argued that HUDI does not to provide any guarantees of correctness if concurrent writers execute the same plan,

I think it still makes sense if the guarantees induces too much complexity, which would confuse a lot.